File: problemas/ad-hoc/2434/2434.cs
using System;
using System.Collections.Generic;
using System.Linq;

class URI {
    static void Main(string[] args) {
        List<int> entrada = Console.ReadLine().Trim().Split(' ').Select(x => int.Parse(x)).ToList();
        int n = entrada[0];
        int saldo = entrada[1];

        int resposta = saldo;
        for(int i = 0; i < n; ++i){
            int movimentacao = int.Parse(Console.ReadLine().Trim());
            saldo += movimentacao;

            resposta = Math.Min(resposta, saldo);
        }

        Console.WriteLine(resposta);
    }
}

File: problemas/ad-hoc/2434/2434.jl
using Printf

entrada = split(readline())
numeros = [parse(Int64, elemento) for elemento in entrada]
n = numeros[1]
saldo = numeros[2]

resposta = saldo
for i = 1:n
    movimentacao = parse(Int64, readline())
    global saldo += movimentacao

    global resposta = min(resposta, saldo)
end

@printf "%d\n" resposta
